src/esp/bindings/bindings.cpp

```diff
@@ -61,15 +61,18 @@ PYBIND11_MODULE(habitat_sim_bindings, m) {
 
   m.import("magnum.scenegraph");
 
+  py::bind_map<std::map<std::string, std::string>>(m, "MapStringString");
+
+  // NOTE(msb) These need to be run in dependency order.
+  // TODO(msb) gfx, scene, and sensor should not cross-depend
+  // TODO(msb) sim and sensor should not cross-depend
   esp::initEspBindings(m);
   esp::core::initCoreBindings(m);
   esp::geo::initGeoBindings(m);
-  esp::gfx::initGfxBindings(m);
-  esp::nav::initShortestPathBindings(m);
   esp::physics::initPhysicsBindings(m);
   esp::scene::initSceneBindings(m);
+  esp::gfx::initGfxBindings(m);
   esp::sensor::initSensorBindings(m);
+  esp::nav::initShortestPathBindings(m);
   esp::sim::initSimBindings(m);
```

Comment:
What could help with untangling is having a separate "declare bindings" step instantiating all the types, and then filling the `py::class` instances doesn't need to be ordered in any particular way.

Reply:
I looked into this and it was Erik's original suggestion. It would mean a lot more boiler plate code. I would prefer to fix the dependency graph (seems doable based on the current set of errors) and then enforce that the dependency graph remains a DAG. I've worked on a few project where we avoided cross-references and it worked out quite well and added a nice discipline. It forced people to think about proper modularity and architecture and resulted in a nice architecture that was easy to understand and reason about.

Reply:
From my experience, even when exposing the basic Magnum math, circular references were inevitable due to templates no longer being templates -- in my case I wanted to expose integer/float vectors in separate compilation units to speed up build times, but there were circular deps coming from conversion constructors (Vector2i->Vector2 and Vector2->Vector2i).

Reply:
I agree. Circular references within a module are inevitable but I think circular references across modules can and should be avoided.

Regardless of which direction we go, I think this change is harmless and independent of what we decide. It's merely shuffling a few declarations around to put things in dependency order.

Approved?

Reply:
I'm not attached to having a DAG but if we can pull it off, we should. It's a nice property to have. If it becomes too restrictive, then we can drop the constraint.

Reply:
Yes, I wasn't opposing this change at all, just sharing my own experience. Sorry if it sounded like a disproval.